libdnf/hy-query.c

```diff
@@ -332,6 +336,62 @@ filter_all(HyQuery q, struct _Filter *f, Map *m)
 }
 
 static void
+filter_name(HyQuery q, struct _Filter *f, Map *m)
+{
+    Pool *pool = dnf_sack_get_pool(q->sack);
+    Id match_name_id = 0;
+    for (int mi = 0; mi < f->nmatches; ++mi) {
+        const char *match = f->matches[mi].str;
+        if (f->cmp_type & HY_EQ)
+            match_name_id = pool_str2id(pool, f->matches[mi].str, 1);
```

Comment:
You can use `match` instead `f->matches[mi].str`
Why last argument is `1` ? Why do you want create new id? Memory is reallocated. And You lose information about `match` existence in pool. Don't do that.